Lib/asyncore.py

```diff
@@ -144,20 +145,20 @@ def poll(timeout=0.0, map=None):
         r, w, e = select.select(r, w, e, timeout)
 
         for fd in r:
-            obj = map.get(fd)
-            if obj is None:
+            obj = map_copy.get(fd)
+            if fd not in map or map.get(fd) != obj:
```

Comment:
We are looking up fd in map twice - we can simplify this to:

    if map.get(fd) is not obj:
        continue

We should handle events on obj only if it is still in the map under the same fd. If it is not in the map (None), or another object is using this fd, we should ignore this object.